prompt_toolkit/key_binding/bindings/vi.py

```diff
@@ -1464,6 +1490,60 @@ def _(event):
         """
         event.current_buffer.insert_text(event.data, overwrite=True)
 
+    @handle(Keys.Backspace, filter=insert_selection_left_mode)
+    @handle(Keys.Delete, filter=insert_selection_left_mode)
+    @handle(Keys.Any, filter=insert_selection_left_mode)
+    def _(event):
+        """
+        Insert data at the beginning of each line of the block selection.
+        """
+        cursor_position = event.current_buffer.cursor_position
+        selection_left_cursor_positions = event.current_buffer.selection_left_cursor_positions
+        insert = False
+        if event.data == '\x7f': # Backspace
+            j = [-1, -1, -1]
+        elif event.data == '\x1b[3~': # Delete
+            j = [0, -1, 0]
+        else: # Otherwise
+            insert = True
+            j = [0, 1, 1]
+        for i1, cp in enumerate(selection_left_cursor_positions):
+            event.current_buffer.cursor_position = cp + (cursor_position - selection_left_cursor_positions[0]) + j[0]
+            if insert:
+                event.current_buffer.insert_text(event.data)
+            else:
+                event.current_buffer.delete()
+            for i2 in range(i1 + 1, len(selection_left_cursor_positions)):
+                selection_left_cursor_positions[i2] += j[1]
+        event.current_buffer.cursor_position = cursor_position + j[2]
+
+    @handle(Keys.Backspace, filter=insert_selection_right_mode)
+    @handle(Keys.Delete, filter=insert_selection_right_mode)
+    @handle(Keys.Any, filter=insert_selection_right_mode)
+    def _(event):
```

Comment:
The code between inserting left and right is identical. I'm going to replace this by a feature I'd call "multiple cursors". Whether the insert positions are coming from the left or right side of the selection doesn't matter. We just have multiple positions where we can insert text.